In [1]:
import pandas as pd

In [2]:
wine_frame = pd.read_csv('./wine-reviews/wine_data_1.csv')

In [3]:
# Seeing what shape the data is in.
wine_frame[:5]

Unnamed: 0 country                                        description  \
0           0      US  This tremendous 100% varietal wine hails from ...   
1           1   Spain  Ripe aromas of fig, blackberry and cassis are ...   
2           2      US  Mac Watson honors the memory of a wine once ma...   
3           3      US  This spent 20 months in 30% new French oak, an...   
4           4  France  This is the top wine from La Bégude, named aft...   

                            designation  points  price        province  \
0                     Martha's Vineyard      96  235.0      California   
1  Carodorum Selección Especial Reserva      96  110.0  Northern Spain   
2         Special Selected Late Harvest      96   90.0      California   
3                               Reserve      96   65.0          Oregon   
4                            La Brûlade      95   66.0        Provence   

            region_1           region_2             variety  \
0        Napa Valley               Napa  Cabernet Sauvignon   
1               Toro                NaN       Tinta de Toro   
2     Knights Valley             Sonoma     Sauvignon Blanc   
3  Willamette Valley  Willamette Valley          Pinot Noir   
4             Bandol                NaN  Provence red blend   

                    winery  
0                    Heitz  
1  Bodega Carmen Rodríguez  
2                 Macauley  
3                    Ponzi  
4     Domaine de la Bégude

In [4]:
# Create a slice that only Contains Califorina Cabernet Savignon
cali_cab =wine_frame[ wine_frame.province == 'California']
cali_cab =cali_cab[cali_cab.variety =='Cabernet Sauvignon']

In [5]:
#How many Califorina Cabernet are in the Data Set
cali_cab_count=cali_cab.country.count()
cali_cab_count 

7452

In [6]:
# What is the average rating?
cali_cab.points.mean()

88.47839506172839

In [7]:
# What is the average price?
cali_cab.price.mean()

52.38737040527804

In [8]:
tim ="timothy"

 There are {{tim}} California Cabs in our data set.

### There are a total of 7452 Californa Cab in our data set, with an average rating of 88.49 and and Average Price of $52.39

In [9]:
# For every California Cab in the Data set, 
# Returns the Rating,
# returns the average price based on Rating, 
# along with the total number of wines with that rating

cali_cab_above_90_avg = []
for x in range (90,101):
    cali_cab_above_90 =cali_cab[cali_cab.points == x]
    cali_cab_above_90_avg.append((x,cali_cab_above_90.price.mean(),cali_cab_above_90.price.count()))
    
cali_cab_above_90_avg   

[(90, 53.57654320987654, 810),
 (91, 58.878406708595385, 477),
 (92, 66.83512064343164, 746),
 (93, 79.36652542372882, 472),
 (94, 91.12602739726027, 365),
 (95, 105.59685863874346, 191),
 (96, 113.68354430379746, 79),
 (97, 163.73809523809524, 42),
 (98, 111.66666666666667, 6),
 (99, 108.33333333333333, 3),
 (100, 205.0, 3)]

In [10]:
cali_cab_99 =cali_cab[ cali_cab.points >= 97]
cali_cab_99

Unnamed: 0 country                                        description  \
14573        14573      US  So good, so rich, so stunning, it's just a joy...   
18617        18617      US  The former basketball star's initial release i...   
23764        23764      US  In keeping with the voluptuous approachability...   
23765        23765      US  One sip is all it takes to fall in love with t...   
24156        24156      US  Pretty much as good as Napa Cabernet gets. Sho...   
24161        24161      US  A very great Napa Cabernet by any standard. De...   
24163        24163      US  A true sensation. Shows the richness and conce...   
26663        26663      US  So delicious now, you can hardly keep your han...   
28958        28958      US  This 100% Cabernet Sauvignon is aged in 85% ne...   
31854        31854      US  This low production, 100% Cab offers the most ...   
34928        34928      US  An outstanding wine, the best California Caber...   
41526        41526      US  Pretty much as good as Napa Cabernet gets. Sho...   
41531        41531      US  A very great Napa Cabernet by any standard. De...   
41533        41533      US  A true sensation. Shows the richness and conce...   
51143        51143      US  So good, so rich, so stunning, it's just a joy...   
52261        52261      US  Price be damned, this is an authentically gorg...   
56727        56727      US  Awesome Cabernet from this esteemed vineyard i...   
60779        60779      US  This wine is from John Schwartz, who partners ...   
65330        65330      US  This is really a spectacular wine. It's hard t...   
71056        71056      US  Beautiful wine, lush and soft in the style of ...   
74057        74057      US  The former basketball star's initial release i...   
78008        78008      US  This 100% Cabernet Sauvignon is aged in 85% ne...   
81688        81688      US  This is Cabernet Sauvignon lifted to greatness...   
81964        81964      US  Tasted blind in a large group of Napa Valley C...   
81965        81965      US  Pretty much as good as any Cabernet out there,...   
82998        82998      US  Another fabulous Cardinale, better than the 20...   
89399        89399      US  Tasted in a flight of great and famous Napa wi...   
89642        89642      US  From Chilean winemaker, Aurelio Montes, this 1...   
90439        90439      US  There are incredible aromatics on this Caberne...   
90440        90440      US  So enormously delicious, it will be hard to ke...   
90442        90442      US  A dramatic wine, this is rich, full bodied and...   
92916        92916      US  A perfect score has to be considered in the co...   
104199      104199      US  This wine will divide aficionados. Some will f...   
106223      106223      US  So good, so rich, so stunning, it's just a joy...   
108683      108683      US  Lancaster burst on the scene with enormous buz...   
108927      108927      US  Awesome Cabernet from this esteemed vineyard i...   
114279      114279      US  The impression is of a young, tannicly closed ...   
116690      116690      US  This is really a spectacular wine. It's hard t...   
119526      119526      US  Pretty much as good as Napa Cabernet gets. Sho...   
119531      119531      US  A very great Napa Cabernet by any standard. De...   
119533      119533      US  A true sensation. Shows the richness and conce...   
121310      121310      US  This is really a spectacular wine. It's hard t...   
127492      127492      US  This is nearly perfect, a massive, monumental ...   
127493      127493      US  A monumental Cabernet that succeeds on every l...   
127495      127495      US  Massive, dramatic, beautiful, intense, but you...   
127497      127497      US  Monumentally detailed, this 100% Cabernet Sauv...   
127608      127608      US  Tasted in a big lineup of St. Helena Cabernets...   
133857      133857      US  Awesome Cabernet from this esteemed vineyard i...   
137099      137099      US  Tasted in a

In [11]:
wash_cab =wine_frame[ wine_frame.province == 'Washington']
wash_cab =wash_cab[wash_cab.variety =='Cabernet Sauvignon']

In [12]:
wash_cab_count=wash_cab.country.count()
wash_cab_count 

1481

In [13]:
# What is the average rating?
wash_cab.points.mean()

89.43079000675219

In [14]:
# What is the average price?
wash_cab.price.mean()

36.64358452138493

In [15]:
# For every California Cab in the Data set, 
# Returns the Rating,
# returns the average price based on Rating, 
# along with the total number of wines with that rating

wash_cab_above_90_avg = []
for x in range (90,101):
    wash_cab_above_90 =wash_cab[wash_cab.points == x]
    wash_cab_above_90_avg.append((x,wash_cab_above_90.price.mean(),wash_cab_above_90.price.count()))
    
wash_cab_above_90_avg  

[(90, 35.44859813084112, 214),
 (91, 40.328859060402685, 149),
 (92, 40.52287581699346, 153),
 (93, 51.922222222222224, 90),
 (94, 54.21666666666667, 60),
 (95, 68.74193548387096, 31),
 (96, 87.9090909090909, 22),
 (97, 75.0909090909091, 11),
 (98, 128.33333333333334, 3),
 (99, 125.0, 1),
 (100, nan, 0)]

In [16]:
wash_cab_99 =wash_cab[ wash_cab.points >= 97]
wash_cab_99

Unnamed: 0 country                                        description  \
22065        22065      US  Sourced from old-vine Bacchus and Weinbau frui...   
34921        34921      US  The flagship wine from Quilceda Creek offers e...   
51178        51178      US  As good as any Père de Famille to date, this b...   
56728        56728      US  This Père de Famille is 85% Cabernet Sauvignon...   
56729        56729      US  The 76% Cabernet Sauvignon is melded with 17% ...   
89640        89640      US  The 2006 flagship Cabernet from Quilceda Creek...   
103997      103997      US  The fragrance is almost otherworldly; sweet, d...   
104197      104197      US  A bold, muscular, striking wine with nerves of...   
104198      104198      US  This is the best bottling of a very fine lineu...   
108928      108928      US  This Père de Famille is 85% Cabernet Sauvignon...   
108929      108929      US  The 76% Cabernet Sauvignon is melded with 17% ...   
133858      133858      US  This Père de Famille is 85% Cabernet Sauvignon...   
133859      133859      US  The 76% Cabernet Sauvignon is melded with 17% ...   
137026      137026      US  Lovely color and aromatics, this supremely pow...   
137370      137370      US  The 2006 flagship Cabernet from Quilceda Creek...   

            designation  points  price    province                 region_1  \
22065           Reserve      97   85.0  Washington     Columbia Valley (WA)   
34921               NaN      99  125.0  Washington     Columbia Valley (WA)   
51178   Père de Famille      97   68.0  Washington     Columbia Valley (WA)   
56728   Père de Famille      97   60.0  Washington     Columbia Valley (WA)   
56729               NaN      97   85.0  Washington  Walla Walla Valley (WA)   
89640               NaN      98  125.0  Washington     Columbia Valley (WA)   
103997  Père de Famille      97   55.0  Washington     Columbia Valley (WA)   
104197              NaN      98  135.0  Washington     Columbia Valley (WA)   
104198              NaN      97   88.0  Washington  Walla Walla Valley (WA)   
108928  Père de Famille      97   60.0  Washington     Columbia Valley (WA)   
108929              NaN      97   85.0  Washington  Walla Walla Valley (WA)   
133858  Père de Famille      97   60.0  Washington     Columbia Valley (WA)   
133859              NaN      97   85.0  Washington  Walla Walla Valley (WA)   
137026              NaN      97   95.0  Washington               Washington   
137370              NaN      98  125.0  Washington     Columbia Valley (WA)   

                region_2             variety           winery  
22065    Columbia Valley  Cabernet Sauvignon            Abeja  
34921    Columbia Valley  Cabernet Sauvignon   Quilceda Creek  
51178    Columbia Valley  Cabernet Sauvignon      Betz Family  
56728    Columbia Valley  Cabernet Sauvignon      Betz Family  
56729    Columbia Valley  Cabernet Sauvignon  Leonetti Cellar  
89640    Columbia Valley  Cabernet Sauvignon   Quilceda Creek  
103997   Columbia Valley  Cabernet Sauvignon      Betz Family  
104197   Columbia Valley  Cabernet Sauvignon   Quilceda Creek  
104198   Columbia Valley  Cabernet Sauvignon  Leonetti Cellar  
108928   Columbia Valley  Cabernet Sauvignon      Betz Family  
108929   Columbia Valley  Cabernet Sauvignon  Leonetti Cellar  
133858   Columbia Valley  Cabernet Sauvignon      Betz Family  
133859   Columbia Valley  Cabernet Sauvignon  Leonetti Cellar  
137026  Washington Other  Cabernet Sauvignon   Quilceda Creek  
137370   Columbia Valley  Cabernet Sauvignon   Quilceda Creek